torchgeo/datasets/idtrees.py

```diff
@@ -383,14 +396,42 @@ def _load_geometries(self, directory: str) -> Dict[int, Dict[str, Any]]:
         features: Dict[int, Dict[str, Any]] = {}
         for path in filepaths:
             with fiona.open(path) as src:
-                for i, feature in enumerate(src):
+                for feature in src:
                     if self.split == "train":
                         features[feature["properties"]["id"]] = feature
                     # Test set task 2 has no id
                     else:
-                        features[i] = feature
+                        id = len(features) + 1
+                        features[id] = feature
         return features
 
+    def _filter_boxes(
```

Comment:
I think the correct way to do this is the unfortunately verbose:
```python
from typing import override

@override
def _filter_boxes(self, image_size: Tuple[int, int], boxes: Tensor, min_size: int, labels: Tensor) -> Tuple[Tensor, Tensor]:
    ...

@override 
def _filter_boxes(self, image_size: Tuple[int, int], boxes: Tensor, min_size: int, labels: None) -> Tuple[Tensor, None]:
    ...
```
followed by the same code below. Basically, if the input labels is Tensor, the output labels is also Tensor, and if it's None it's None. Then you can remove the `type: ignore` above.